### PR TITLE
Added changes for sending targeted meeting notifications

### PR DESCRIPTION
**Pull request by Prithvi-MSFT**

### CODE REVIEW

**Comment by ChetanSharma-msft** [comment]
@Prithvi-MSFT 

**Make sure that you have verified below points in your PR:** 
1) Add build pipeline for C#/NodeJS samples.
2) Add deployment pipeline for every new sample like for C#, if sample exist otherwise create it for NodeJS sample.
3) Deploy the sample manually, test it and add the deployed manifest in your sample folder.
4) Make sure that you have added the deployed manifest link in your Readme file.
5) Make sure to link your sample with main/root readme.
6) Make sure that you have linked the proper Microsoft Documentation with your sample's readme file in "Further reading" section.
7) Once sample is pushed, make sure to raise a doc bug on Microsoft documentation to link your sample on that page.
8) Make sure that you have added sample.json with proper links of .gif and metadata.
9) Make sure to run build pipeline every time you push the new changes in PR, including PR comments fixes and add the screenshot/link with your PR.
10) Verify that your readme file is visible properly in preview section. 

Code related guidelines:
1) Remove all extra spaces and lines from your code.
2) Make sure that you have added proper summary for each and every methods.
3) Don't push credentials in your PR.
4) Verify alignments and correct those.
5) Don't put any hardcoded value and push those in PR.
6) Remove unwanted commented code
7) Use latest framework and tools for your code implementation

File: samples/meetings-content-bubble/csharp/ContentBubble/Bots/ContentBubbleBot.cs [hunk 1]
```diff
@@ -34,21 +42,57 @@ public ContentBubbleBot(IConfiguration configuration)
                                         new AgendaItem { Topic = "Increase research budget by 10%" , Id = 2},
                                         new AgendaItem { Topic = "Continue with WFH for next 3 months" , Id = 3}},
             };
+            _httpClientFactory = httpClientFactory;
         }
         protected override async Task OnMessageActivityAsync(ITurnContext<IMessageActivity> turnContext, CancellationToken cancellationToken)
```
Comment (ChetanSharma-msft) [inline samples/meetings-content-bubble/csharp/ContentBubble/Bots/ContentBubbleBot.cs:47]: One line space before it & also add method summary

File: samples/meetings-content-bubble/csharp/ContentBubble/Models/MeetingNotification.cs [hunk 2]
```diff
@@ -0,0 +1,12 @@
+﻿using System.Collections.Generic;
```
Comment (ChetanSharma-msft) [inline samples/meetings-content-bubble/csharp/ContentBubble/Models/MeetingNotification.cs:1]: Add file header

File: samples/meetings-content-bubble/csharp/ContentBubble/Models/ActionBase.cs [hunk 3]
```diff
@@ -7,6 +7,14 @@ namespace Content_Bubble_Bot.Models
 {
     public class ActionBase
     {
+        /// <summary>
```
Comment (ChetanSharma-msft) [inline samples/meetings-content-bubble/csharp/ContentBubble/Models/ActionBase.cs:10]: Add file header

File: samples/meetings-content-bubble/csharp/ContentBubble/Models/ParticipantDetail.cs [hunk 4]
```diff
@@ -0,0 +1,16 @@
+﻿namespace Content_Bubble_Bot.Models
```
Comment (ChetanSharma-msft) [inline samples/meetings-content-bubble/csharp/ContentBubble/Models/ParticipantDetail.cs:1]: Add file header

File: samples/meetings-content-bubble/csharp/ContentBubble/Pages/SendNotificationPage.cshtml.cs [hunk 5]
```diff
@@ -0,0 +1,11 @@
+using Microsoft.AspNetCore.Mvc.RazorPages;
```
Comment (ChetanSharma-msft) [inline samples/meetings-content-bubble/csharp/ContentBubble/Pages/SendNotificationPage.cshtml.cs:1]: Add file header

File: samples/meetings-content-bubble/csharp/README.md [hunk 6]
```diff
@@ -90,6 +90,13 @@ Message the Bot by @ mentioning to interact with the content bubble.
...
+## Send targeted meeting notification
+![Meeting card](ContentBubble/Images/MeetingCard.png)
+
+![Target notification](ContentBubble/Images/TargetNotification.png)
+
+
 ## Further Reading
 
-[Meetings-content-bubble](https://techcommunity.microsoft.com/t5/microsoft-teams/introducing-chat-bubbles-in-microsoft-teams-meetings/m-p/2447271)
+[Meetings-content-bubble](https://techcommunity.microsoft.com/t5/microsoft-teams/introducing-chat-bubbles-in-microsoft-teams-meetings/m-p/2447271)
```
Comment (ChetanSharma-msft) [inline samples/meetings-content-bubble/csharp/README.md:102]: Why we removed hyphen here ?

**Review by ChetanSharma-msft — CHANGES_REQUESTED** [review]
Added my comments

File: samples/meetings-content-bubble/csharp/ContentBubble/Bots/ContentBubbleBot.cs (hunk 1 above)
Comment (Prithvi-MSFT) [reply samples/meetings-content-bubble/csharp/ContentBubble/Bots/ContentBubbleBot.cs:57]: Added

File: samples/meetings-content-bubble/csharp/ContentBubble/Models/MeetingNotification.cs (hunk 2 above)
Comment (Prithvi-MSFT) [reply samples/meetings-content-bubble/csharp/ContentBubble/Models/MeetingNotification.cs]: Done

File: samples/meetings-content-bubble/csharp/ContentBubble/Models/ActionBase.cs (hunk 3 above)
Comment (Prithvi-MSFT) [reply samples/meetings-content-bubble/csharp/ContentBubble/Models/ActionBase.cs:10]: Done

File: samples/meetings-content-bubble/csharp/ContentBubble/Pages/SendNotificationPage.cshtml.cs (hunk 5 above)
Comment (Prithvi-MSFT) [reply samples/meetings-content-bubble/csharp/ContentBubble/Pages/SendNotificationPage.cshtml.cs:6]: Done

File: samples/meetings-content-bubble/csharp/README.md (hunk 6 above)
Comment (Prithvi-MSFT) [reply samples/meetings-content-bubble/csharp/README.md]: Added

File: samples/meetings-content-bubble/csharp/ContentBubble/Models/ParticipantDetail.cs (hunk 4 above)
Comment (Prithvi-MSFT) [reply samples/meetings-content-bubble/csharp/ContentBubble/Models/ParticipantDetail.cs]: Added

**Comment by Prithvi-MSFT** [comment]
Build Status - All builds passed

image

https://github.com/OfficeDev/Microsoft-Teams-Samples/actions/runs/4015214232

File: samples/meetings-content-bubble/csharp/ContentBubble/Bots/ContentBubbleBot.cs [hunk 7]
```diff
@@ -23,8 +32,9 @@ public class ContentBubbleBot : TeamsActivityHandler
     {
         private readonly IConfiguration _config;
         private readonly MeetingAgenda _agenda;
+        private IHttpClientFactory _httpClientFactory;
 
-        public ContentBubbleBot(IConfiguration configuration)
+        public ContentBubbleBot(IConfiguration configuration, IHttpClientFactory httpClientFactory)
```
Comment (YunnyChung) [inline samples/meetings-content-bubble/csharp/ContentBubble/Bots/ContentBubbleBot.cs:37]: Let's update the name of the bot/file to be TargetedInMeetingNotifiationBot :) We are intentionally moving away from "ContentBubble"

File: samples/meetings-content-bubble/csharp/assets/sample.json [hunk 8]
```diff
@@ -3,10 +3,10 @@
     "name": "officedev-microsoft-teams-samples-meetings-content-bubble-csharp",
     "source": "officeDev",
     "title": "Content Bubble Meeting app",
-    "shortDescription": "Illustrates how to implement the \"content bubble\" in-meeting experience",
+    "shortDescription": "Illustrates how to implement the \"content bubble\" and \"targeted notifications\" in-meeting experience",
```
Comment (YunnyChung) [inline samples/meetings-content-bubble/csharp/assets/sample.json:6]: I think the description is slightly confusing :) We are rebranding the existing "content bubble" to be "in-meeting notification". This feature is to send notification to everyone in the meeting. 

For the new feature, we want to send the in-meeting notification only to targeted participants, and this bot is demonstrating this feature :) We call this feature as 'targeted in-meeting notification'. 

With these feature names, can we update the description?

File: samples/meetings-content-bubble/csharp/ContentBubble/Bots/ContentBubbleBot.cs [hunk 9]
```diff
@@ -76,23 +137,78 @@ private async Task HandleActions(ITurnContext<IMessageActivity> turnContext)
...
                     };
                     await turnContext.SendActivityAsync(activity);
                     break;
                 case "SubmitFeedback":
-                    var submitFeedback = Newtonsoft.Json.JsonConvert.DeserializeObject<SubmitFeedbackAction>(turnContext.Activity.Value.ToString());
+                    var submitFeedback = JsonConvert.DeserializeObject<SubmitFeedbackAction>(turnContext.Activity.Value.ToString());
                     var item = _agenda.AgendaItems.First(a => a.Id.ToString() == submitFeedback.Choice);
                     await turnContext.SendActivityAsync($"{turnContext.Activity.From.Name} voted **{submitFeedback.Feedback}** for '{item.Topic}'");
                     break;
+                case "SendNotification":
```
Comment (YunnyChung) [inline samples/meetings-content-bubble/csharp/ContentBubble/Bots/ContentBubbleBot.cs:152]: How about we updating the command name to be "SendTargetedMeetingNotification" or something that contains "targeted"?

File: samples/meetings-content-bubble/csharp/ContentBubble/Bots/ContentBubbleBot.cs [hunk 10]
```diff
@@ -76,23 +137,78 @@ private async Task HandleActions(ITurnContext<IMessageActivity> turnContext)
...
+                        var selectedMembers = actionSet.Choice;
+                        var recipients = JsonConvert.SerializeObject(selectedMembers.Split(","));
+                        var pageUrl = JsonConvert.SerializeObject(_config["BaseUrl"] + "/SendNotificationPage");
+                        var meetingId = turnContext.Activity.TeamsGetMeetingInfo()?.Id ?? throw new InvalidOperationException("This method is only valid within the scope of a MS Teams Meeting.");
+
+                        // Notification payload for meeting target notification API.
+                        string notificationPayload = @"{
+                                 ""type"": ""targetedMeetingNotification"",
+                                 ""value"": {
+                                 ""recipients"": " + recipients + @",
```
Comment (YunnyChung) [inline samples/meetings-content-bubble/csharp/ContentBubble/Bots/ContentBubbleBot.cs:165]: I really like the example bot's sceanrio btw! 
(1) Returning the list or participants who are in the meeting 
(2) Allowing a user to choose some participants among them to send targeted in-meeting notifications. 
(3) And each targeted in-meeting notifciation shows a task module. 

I think this whole scenario captures the point of the feature really well. Thank you! 🙏🎉👍

**Review by YunnyChung — COMMENTED** [review]
Leaving a few small comments :)

File: samples/meetings-content-bubble/csharp/ContentBubble/Bots/ContentBubbleBot.cs (hunk 9 above)
Comment (Prithvi-MSFT) [reply samples/meetings-content-bubble/csharp/ContentBubble/Bots/ContentBubbleBot.cs]: Yes updated it

File: samples/meetings-content-bubble/csharp/assets/sample.json (hunk 8 above)
Comment (Prithvi-MSFT) [reply samples/meetings-content-bubble/csharp/assets/sample.json:6]: Sure, updated the description

File: samples/meetings-content-bubble/csharp/ContentBubble/Bots/ContentBubbleBot.cs (hunk 10 above)
Comment (Prithvi-MSFT) [reply samples/meetings-content-bubble/csharp/ContentBubble/Bots/ContentBubbleBot.cs]: Thank You !!

File: samples/meetings-content-bubble/csharp/ContentBubble/Bots/ContentBubbleBot.cs (hunk 7 above)
Comment (Prithvi-MSFT) [reply samples/meetings-content-bubble/csharp/ContentBubble/Bots/ContentBubbleBot.cs]: Yes, updated the file name

File: samples/meetings-content-bubble/csharp/TargetedNotifications/Bots/TargetedInNotificationMeetingBot.cs [hunk 11]
```diff
@@ -0,0 +1,266 @@
...
+        {
+            _config = configuration;
+            _agenda = new MeetingAgenda
+            {
+                AgendaItems = new List<AgendaItem>() {
+                                        new AgendaItem { Topic = "Approve 5% dividend payment to shareholders" , Id = 1 },
+                                        new AgendaItem { Topic = "Increase research budget by 10%" , Id = 2},
+                                        new AgendaItem { Topic = "Continue with WFH for next 3 months" , Id = 3}},
+            };
+            _httpClientFactory = httpClientFactory;
```
Comment (ChetanSharma-msft) [inline samples/meetings-content-bubble/csharp/TargetedNotifications/Bots/TargetedInNotificationMeetingBot.cs:47]: Add one line space before it.

File: samples/meetings-content-bubble/csharp/TargetedNotifications/Bots/TargetedInNotificationMeetingBot.cs [hunk 12]
```diff
@@ -0,0 +1,266 @@
...
+        public TargetedInNotificationMeetingBot(IConfiguration configuration, IHttpClientFactory httpClientFactory)
+        {
+            _config = configuration;
+            _agenda = new MeetingAgenda
+            {
+                AgendaItems = new List<AgendaItem>() {
+                                        new AgendaItem { Topic = "Approve 5% dividend payment to shareholders" , Id = 1 },
+                                        new AgendaItem { Topic = "Increase research budget by 10%" , Id = 2},
+                                        new AgendaItem { Topic = "Continue with WFH for next 3 months" , Id = 3}},
+            };
```
Comment (ChetanSharma-msft) [inline samples/meetings-content-bubble/csharp/TargetedNotifications/Bots/TargetedInNotificationMeetingBot.cs:46]: Alignment doesn't looks correct.

File: samples/meetings-content-bubble/csharp/TargetedNotifications/Manifest/manifest.json [hunk 13]
```diff
@@ -0,0 +1,75 @@
...
+        }
+      ]
+    }
+  ],
+  "permissions": [
+    "identity",
+    "messageTeamMembers"
+  ],
+  "validDomains": [
+    "*.ngrok.io"
```
Comment (ChetanSharma-msft) [inline samples/meetings-content-bubble/csharp/TargetedNotifications/Manifest/manifest.json:59]: Please add placeholder here as well.

File: samples/meetings-content-bubble/csharp/TargetedNotifications/Pages/Shared/_Layout.cshtml [hunk 14]
```diff
@@ -0,0 +1,19 @@
+﻿<!DOCTYPE html>
+<html lang="en">
+<head>
+    <title>Microsoft Teams Meeting content bubble sample</title>
+
```
Comment (ChetanSharma-msft) [inline samples/meetings-content-bubble/csharp/TargetedNotifications/Pages/Shared/_Layout.cshtml:5]: Should we change this name as well?
content bubble sample

**Review by ChetanSharma-msft — CHANGES_REQUESTED** [review]
Added few comments

File: samples/meetings-content-bubble/csharp/TargetedNotifications/Bots/TargetedInNotificationMeetingBot.cs (hunk 11 above)
Comment (Prithvi-MSFT) [reply samples/meetings-content-bubble/csharp/TargetedNotifications/Bots/TargetedInNotificationMeetingBot.cs]: changed

File: samples/meetings-content-bubble/csharp/TargetedNotifications/Bots/TargetedInNotificationMeetingBot.cs (hunk 12 above)
Comment (Prithvi-MSFT) [reply samples/meetings-content-bubble/csharp/TargetedNotifications/Bots/TargetedInNotificationMeetingBot.cs:49]: changed

File: samples/meetings-content-bubble/csharp/TargetedNotifications/Pages/Shared/_Layout.cshtml (hunk 14 above)
Comment (Prithvi-MSFT) [reply samples/meetings-content-bubble/csharp/TargetedNotifications/Pages/Shared/_Layout.cshtml:5]: changed

File: samples/meetings-content-bubble/csharp/TargetedNotifications/Manifest/manifest.json (hunk 13 above)
Comment (Prithvi-MSFT) [reply samples/meetings-content-bubble/csharp/TargetedNotifications/Manifest/manifest.json]: added

**Review by ChetanSharma-msft — APPROVED** [review]
Changes looks fine, approving it.

**Comment by ChetanSharma-msft** [comment]
Executed build pipeline: https://github.com/OfficeDev/Microsoft-Teams-Samples/actions/runs/4027088862

**Comment by Prithvi-MSFT** [comment]
Executed build pipeline 
status - all builds passed
https://github.com/OfficeDev/Microsoft-Teams-Samples/actions/runs/4030946197

**Review by Wajeed-msft — APPROVED** [review]

File: samples/meetings-content-bubble/csharp/TargetedNotifications/Bots/TargetedInNotificationMeetingBot.cs [hunk 15]
```diff
@@ -0,0 +1,268 @@
...
+using System.Net.Http;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using System.Web;
+
+
+namespace TargetedInNotificationMeetingBot
+{
+    public class TargetedInNotificationMeetingBot : TeamsActivityHandler
```
Comment (YunnyChung) [inline samples/meetings-content-bubble/csharp/TargetedNotifications/Bots/TargetedInNotificationMeetingBot.cs:31]: A minor feedback: Can we update the name to be `TargetedInMeetingNotificationBot` or `TargetedNotificationMeetingBot` as "targeted in meeting notification" / "Targeted Meeting Notification" is the feature name on the public doc?  And let's be consistent with that name throughout the code.

File: samples/meetings-content-bubble/csharp/README.md [hunk 16]
```diff
@@ -13,13 +13,13 @@ extensions:
 urlFragment: officedev-microsoft-teams-samples-meetings-content-bubble-csharp
 ---
 
-# Meetings Content Bubble
+# Meeting Targeted Notification
```
Comment (YunnyChung) [inline samples/meetings-content-bubble/csharp/README.md:16]: Minor: Let's be coherent with the public doc's name of the feature :)
Targeted In-Meeting Notification

File: samples/meetings-content-bubble/csharp/README.md [hunk 17]
```diff
@@ -13,13 +13,13 @@ extensions:
...
 
-This sample illustrates how to implement [Content Bubble](https://docs.microsoft.com/en-us/microsoftteams/platform/apps-in-teams-meetings/create-apps-for-teams-meetings?view=msteams-client-js-latest&tabs=dotnet#notificationsignal-api) In-Meeting Experience.
+This sample illustrates how to implement [Content Bubble](https://docs.microsoft.com/en-us/microsoftteams/platform/apps-in-teams-meetings/create-apps-for-teams-meetings?view=msteams-client-js-latest&tabs=dotnet#notificationsignal-api) In-Meeting Experience. This app also showcase sending targeted notification to meeting participants.
```
Comment (YunnyChung) [inline samples/meetings-content-bubble/csharp/README.md:18]: Minor: I see [ContentBubble]. Let's update it to be '[Targeted In-Meeting Notification]' And Let's also update the link to the feature. This is the new one: https://review.learn.microsoft.com/en-us/microsoftteams/platform/apps-in-teams-meetings/meeting-apps-apis?branch=pr-en-us-7615&tabs=dotnet#targeted-meeting-notification-api

File: samples/meetings-content-bubble/csharp/TargetedNotifications/Pages/ContentBubble.cshtml.cs [hunk 18]
```diff
@@ -1,11 +1,12 @@
...
+// Licensed under the MIT License.
+// </copyright>
+
 using Microsoft.AspNetCore.Mvc;
 using Microsoft.AspNetCore.Mvc.RazorPages;
 
-namespace Content_Bubble_Bot.Pages
+namespace TargetedInNotificationMeetingBot.Pages
 {
     public class ContentBubbleModel : PageModel
```
Comment (YunnyChung) [inline samples/meetings-content-bubble/csharp/TargetedNotifications/Pages/ContentBubble.cshtml.cs:11]: Is this `contentBubbleModel` being used for in-meeting notification scenario (the existing feature) or targeted in-meeting notification scenario (the new feature)? 

Depending on which feature this file is for, can we update the name accordingly? 
ex: If this model class is for the in-meeting notification, we can update the name to be InMeetingNotificationModel. 

If this model classi s for the targeted in-meeting notification, we can update the name to be TargetedInMeetingNotificationModel or TargetedMeetingNotification Model. 

Same question for ContentBubble.cshtml file name as well :)

More about the features:
In-meeting notification (previously called, content bubble): https://review.learn.microsoft.com/en-us/microsoftteams/platform/apps-in-teams-meetings/meeting-apps-apis?branch=pr-en-us-7615&tabs=dotnet#send-an-in-meeting-notification

Targeted Meeting notification: https://review.learn.microsoft.com/en-us/microsoftteams/platform/apps-in-teams-meetings/meeting-apps-apis?branch=pr-en-us-7615&tabs=dotnet#targeted-meeting-notification-api

File: samples/meetings-content-bubble/csharp/TargetedNotifications/Pages/SendNotificationPage.cshtml [hunk 19]
```diff
@@ -0,0 +1,30 @@
...
+@model TargetedInNotificationMeetingBot.Pages.SendNotificationPage
+@{
+    ViewData["Title"] = "Layout";
+}
+
+<script>
+    microsoftTeams.app.initialize();
+    function validateForm() {
+        var feedbackInfo = {
+            Topic: 'Meeting Targeted Notification',
```
Comment (YunnyChung) [inline samples/meetings-content-bubble/csharp/TargetedNotifications/Pages/SendNotificationPage.cshtml:11]: Minor: Targeted Meeting Notification or Targeted In-Meeting Notification

File: samples/meetings-content-bubble/csharp/assets/sample.json (hunk 8 above)
Comment (YunnyChung) [reply samples/meetings-content-bubble/csharp/assets/sample.json:6]: Hey @Prithvi-MSFT , I still see "content bubble" here :) 
This is the description that I am thinking. 
"Illustrates how to implement the \"in-meeting notification\" and \"targeted meeting notification\" experience".

In-meeting notification (previously called, content bubble): https://review.learn.microsoft.com/en-us/microsoftteams/platform/apps-in-teams-meetings/meeting-apps-apis?branch=pr-en-us-7615&tabs=dotnet#send-an-in-meeting-notification

Targeted Meeting notification: https://review.learn.microsoft.com/en-us/microsoftteams/platform/apps-in-teams-meetings/meeting-apps-apis?branch=pr-en-us-7615&tabs=dotnet#targeted-meeting-notification-api

File: samples/meetings-content-bubble/csharp/assets/sample.json [hunk 20]
```diff
@@ -3,10 +3,10 @@
...
     "url": "https://github.com/OfficeDev/Microsoft-Teams-Samples/tree/main/samples/meetings-content-bubble/csharp",
     "longDescription": [
-      "Illustrates how to implement the \"content bubble\" in-meeting experience"
+      "Illustrates how to implement the \"targeted in-meeting notifications\" in-meeting experience"
```
Comment (YunnyChung) [inline samples/meetings-content-bubble/csharp/assets/sample.json:9]: Minor: This seems a bit weird as we are using in-meeting twice. Maybe we can update it to be
"Illustrate how to implement the \"targeted in-meeting notifications\" experience" or 
"Illustrate how to implemenet the \"targeted meeting notifications\" in meetings".

**Review by YunnyChung — COMMENTED** [review]
Minor comments regarding to file/class names :)

File: samples/meetings-content-bubble/csharp/assets/sample.json (hunk 20 above)
Comment (Prithvi-MSFT) [reply samples/meetings-content-bubble/csharp/assets/sample.json]: Yes updated it

File: samples/meetings-content-bubble/csharp/TargetedNotifications/Pages/SendNotificationPage.cshtml (hunk 19 above)
Comment (Prithvi-MSFT) [reply samples/meetings-content-bubble/csharp/TargetedNotifications/Pages/SendNotificationPage.cshtml]: Changed this

File: samples/meetings-content-bubble/csharp/TargetedNotifications/Pages/ContentBubble.cshtml.cs (hunk 18 above)
Comment (Prithvi-MSFT) [reply samples/meetings-content-bubble/csharp/TargetedNotifications/Pages/ContentBubble.cshtml.cs]: Updated this based on page model

File: samples/meetings-content-bubble/csharp/README.md (hunk 17 above)
Comment (Prithvi-MSFT) [reply samples/meetings-content-bubble/csharp/README.md]: Updated this

File: samples/meetings-content-bubble/csharp/README.md (hunk 16 above)
Comment (Prithvi-MSFT) [reply samples/meetings-content-bubble/csharp/README.md]: Updated this

File: samples/meetings-content-bubble/csharp/TargetedNotifications/Bots/TargetedInNotificationMeetingBot.cs (hunk 15 above)
Comment (Prithvi-MSFT) [reply samples/meetings-content-bubble/csharp/TargetedNotifications/Bots/TargetedInNotificationMeetingBot.cs]: Updated the classes at all relevant places

File: samples/meetings-notification/csharp/TargetedNotifications/Bots/TargetedInMeetingNotificationBot .cs [hunk 21]
```diff
@@ -0,0 +1,268 @@
...
+using System.Net.Http;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using System.Web;
+
+
+namespace TargetedInMeetingNotificationBot
+{
+    public class TargetedInNotificationMeetingBot : TeamsActivityHandler
```
Comment (YunnyChung) [inline samples/meetings-notification/csharp/TargetedNotifications/Bots/TargetedInMeetingNotificationBot .cs:31]: I think the class name & the constructor name needs to be updated as well :)

File: samples/meetings-notification/csharp/TargetedNotifications/Models/AgendaItem.cs [hunk 22]
```diff
@@ -1,4 +1,4 @@
-﻿namespace Content_Bubble_Bot.Models
+﻿namespace TargetedInMeetingNotificationBot.Models
```
Comment (YunnyChung) [inline samples/meetings-notification/csharp/TargetedNotifications/Models/AgendaItem.cs:1]: Thanks for updating everything here! 🙏

**Review by YunnyChung — COMMENTED** [review]
Just one small comment as we missed one thing :) Otherwise everything looks good to me 👍

File: samples/meetings-notification/csharp/TargetedNotifications/Bots/TargetedInMeetingNotificationBot .cs (hunk 21 above)
Comment (Prithvi-MSFT) [reply samples/meetings-notification/csharp/TargetedNotifications/Bots/TargetedInMeetingNotificationBot .cs]: Yes, updated

**Review by YunnyChung — APPROVED** [review]

**Comment by Prithvi-MSFT** [comment]
Build status - All builds passed.
https://github.com/OfficeDev/Microsoft-Teams-Samples/actions/runs/4046829542

**Pull request merged**